### PR TITLE
Replaced the legacy java.util.Date with Instant

### LINKED ISSUE #30: Replace the legacy java.util.Date with Instant or LocalDate
Suggestion: https://twitter.com/p_chmielowski/status/1560027370874085376?s=20&t=DDL0ltrfh6q-JieM_KESIw

### PR DESCRIPTION
**Pull request by sravank9**
Closes #30

### CODE REVIEW

File: app/src/main/java/com/waseefakhtar/doseapp/data/Converters.kt [hunk 1]
```diff
@@ -13,11 +16,23 @@ class Converters {
         return value?.let { Date(it) }
     }
 
+    @RequiresApi(Build.VERSION_CODES.O)
+    @TypeConverter
+    fun fromInstant(value: Long?): Instant? {
+        return value?.let { Instant.ofEpochMilli(it) }
+    }
+
```
Comment (waseefakhtar) [inline app/src/main/java/com/waseefakhtar/doseapp/data/Converters.kt:24]: Will this conversion only work for Android O and later? Perhaps add a check and conversion for pre Android O as well?

File: app/src/main/java/com/waseefakhtar/doseapp/data/Converters.kt [hunk 2]
```diff
@@ -13,11 +16,23 @@ class Converters {
...
     fun dateToTimestamp(date: Date?): Long? {
         return date?.time
     }
 
+    @RequiresApi(Build.VERSION_CODES.O)
+    @TypeConverter
+    fun instantToTimeStamp(instant: Instant?): Long? {
+        return instant?.toEpochMilli()
+    }
+
```
Comment (waseefakhtar) [inline app/src/main/java/com/waseefakhtar/doseapp/data/Converters.kt:35]: Same here.

**Review by waseefakhtar — COMMENTED** [review]
Looks good to me! I just had a question about compatibility of type converter in pre Android O.

File: app/src/main/java/com/waseefakhtar/doseapp/data/Converters.kt (hunk 1 above)
Comment (Wottrich) [reply app/src/main/java/com/waseefakhtar/doseapp/data/Converters.kt:24]: `Instant` has compatibility with API 26 or later. Looking your minSdk, this change may not have sense for now, in my point of view.

Comment (waseefakhtar) [reply app/src/main/java/com/waseefakhtar/doseapp/data/Converters.kt:24]: I agree @Wottrich!

**Pull request closed without merging**